scripts/bin/set_environment

```diff
@@ -0,0 +1,130 @@
+#!/bin/ruby
```

Comment:
Can we try to run this using a shebang pointing to `/opt/puppetlabs/puppet/bin/ruby`?

Reply:
I was unable to get that to work on our dev environment, when I was working on setting up R10K.  The script worked when called outside of the Puppet server, but failed when called by the Puppet server.  I'll revisit to see if I can figure out what the issue was.

Reply:
The problem on our dev Puppet master is that many of the gems in /opt/puppetlabs/puppet are not accessible to puppet.  So, in the puppet server log, you get something like:
 WARN  [c.p.p.ShellUtils] Executed an external process which logged to STDERR: /opt/puppetlabs/puppet/lib/ruby/2.1.0/rubygems/stub_specification.rb:71:in `initialize': Permission denied @ rb_sysopen - /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/specifications/blankslate-2.1.2.4.gemspec (Errno::EACCES)

Reply:
For now, I think it's fine to leave it as `#!/bin/ruby`. There may be a future ticket where we are interested in not installing system ruby by default.

Reply:
The permissions problems are probably that someone forgot to change their `umask` before installing the gems.